src/org/rocketproplab/marginalstability/flightcomputer/comm/GPSTransceiver.java

```diff
@@ -28,8 +30,9 @@ public void onSerialData(String data) {
     if (packet.isValid()) {
       router.recivePacket(packet, PacketSources.GPS);
     } else {
-      System.out.println("Got invalid packet " + data + "!");
-      // TODO report error
+      ErrorReporter errorReporter = ErrorReporter.getInstance();
+      String errorMsg = "Got invalid packet \" + data + \"!\"";
```

Comment:
This should combine the string "Got Invalid packet " with data and "!". Right now it prints the literal String 'Got invalid packet " + data "!"'.